CMakeLists.txt

```diff
@@ -263,6 +263,14 @@ if(CMAKE_COMPILER_IS_CLANG)
     set(CMAKE_C_FLAGS_TSAN        "-fsanitize=thread -O3")
     set(CMAKE_C_FLAGS_TSANDBG     "-fsanitize=thread -O1 -g3 -fno-omit-frame-pointer -fno-optimize-sibling-calls")
     set(CMAKE_C_FLAGS_CHECK       "-Os")
+
+    set(CMAKE_C_FLAGS_CFMEMSAN   "${CMAKE_C_FLAGS_MEMSAN} -DMBEDTLS_TEST_CONSTANT_FLOW_MEMSAN")
```

Comment:
Rather than add new build types for constant-flow msan, the right thing is probably to have CMake detect whether `MBEDTLS_TEST_CONSTANT_FLOW_MEMSAN` is defined, and if so (and if the compiler needs it) add the extra `-f` flag.

(Maybe we should even enable MSan automatically based on `MBEDTLS_TEST_CONSTANT_FLOW_MEMSAN`?)

The way to do that is with [`CheckCSourceCompiles`](https://cmake.org/cmake/help/latest/module/CheckCSourceCompiles.html):

```
include(CheckCSourceCompiles)
CheckCSourceCompiles("
#include <mbedtls/build_info.h>
#if !defined(MBEDTLS_TEST_CONSTANT_FLOW_MEMSAN)
#error MBEDTLS_TEST_CONSTANT_FLOW_MEMSAN not defined
#endif
"
    MBEDTLS_TEST_CONSTANT_FLOW_MEMSAN)
if(MBEDTLS_TEST_CONSTANT_FLOW_MEMSAN)
    …
endif()
```

Reply:
Actually Bence notes that [`CheckSymbolExists`](https://cmake.org/cmake/help/latest/module/CheckSymbolExists.html) also works for macros.

```
include(CheckSymbolExists)
set(CMAKE_REQUIRED_INCLUDES …) # TODO: set this and create a scope for it
check_symbol_exists(MBEDTLS_TEST_CONSTANT_FLOW_MEMSAN
    "mbedtls/build_info.h"
    MBEDTLS_TEST_CONSTANT_FLOW_MEMSAN)
```

Reply:
But that's not quite right because AFAICT it doesn't honor `CFLAGS`, so it could get the wrong configuration if `CFLAGS` includes e.g. `-DMBEDTLS_CONFIG_FILE`.

Reply:
> Rather than add new build types for constant-flow msan, the right thing is probably to have CMake detect whether `MBEDTLS_TEST_CONSTANT_FLOW_MEMSAN` is defined, and if so (and if the compiler needs it) add the extra `-f` flag.

Is it really the right thing? I quite liked having separate build types, what's the problem with that?

Reply:
I'm thinking of applications of the cmake training we just took. But I don't intend to change the approach here in the short term, it's working and I have higher priorities.